pkg/controller/docdb/dbcluster/dbcluster.go

```diff
@@ -0,0 +1,304 @@
+	e.preDelete = preDelete
+	e.filterList = filterList
+}
+
+type hooks struct {
+	client docdbiface.DocDBAPI
+	kube   client.Client
+}
+
+type tagger struct {
+	kube client.Client
+}
+
+func preObserve(_ context.Context, cr *svcapitypes.DBCluster, obj *svcsdk.DescribeDBClustersInput) error {
+	obj.DBClusterIdentifier = awsclient.String(meta.GetExternalName(cr))
+	return nil
+}
+
+func postObserve(_ context.Context, cr *svcapitypes.DBCluster, resp *svcsdk.DescribeDBClustersOutput, obs managed.ExternalObservation, err error) (managed.ExternalObservation, error) {
+	if err != nil {
+		return managed.ExternalObservation{}, err
+	}
+
+	switch awsclient.StringValue(cr.Status.AtProvider.Status) {
+	case svcapitypes.DocDBInstanceStateAvailable:
+		cr.Status.SetConditions(xpv1.Available())
+	case svcapitypes.DocDBInstanceStateCreating:
+		cr.Status.SetConditions(xpv1.Creating())
+	case svcapitypes.DocDBInstanceStateDeleting:
+		cr.Status.SetConditions(xpv1.Deleting())
+	default:
+		cr.Status.SetConditions(xpv1.Unavailable())
+	}
+
+	return obs, nil
+}
+
+func (e *hooks) isUpToDate(cr *svcapitypes.DBCluster, resp *svcsdk.DescribeDBClustersOutput) (bool, error) { // nolint:gocyclo
+	cluster := resp.DBClusters[0]
+
+	if (cr.Spec.ForProvider.BackupRetentionPeriod != nil && !awsclient.IsEqualInt64(cr.Spec.ForProvider.BackupRetentionPeriod, cluster.BackupRetentionPeriod)) ||
+		!awsclient.IsEqualString(cr.Spec.ForProvider.DBClusterIdentifier, cluster.DBClusterIdentifier) ||
+		(cr.Spec.ForProvider.DBClusterParameterGroupName != nil && !awsclient.IsEqualString(cr.Spec.ForProvider.DBClusterParameterGroupName, cluster.DBClusterParameterGroup)) ||
+		(cr.Spec.ForProvider.DBSubnetGroupName != nil && !awsclient.IsEqualString(cr.Spec.ForProvider.DBSubnetGroupName, cluster.DBSubnetGroup)) ||
+		(cr.Spec.ForProvider.DeletionProtection != nil && !awsclient.IsEqualBool(cr.Spec.ForProvider.DeletionProtection, cluster.DeletionProtection)) ||
+		(cr.Spec.ForProvider.EnableCloudwatchLogsExports != nil && !areSameElements(cr.Spec.ForProvider.EnableCloudwatchLogsExports, cluster.EnabledCloudwatchLogsExports)) ||
+		(cr.Spec.ForProvider.Port != nil && !awsclient.IsEqualInt64(cr.Spec.ForProvider.Port, cluster.Port)) ||
+		(cr.Spec.ForProvider.PreferredBackupWindow != nil && !awsclient.IsEqualString(cr.Spec.ForProvider.PreferredBackupWindow, cluster.PreferredBackupWindow)) ||
+		(cr.Spec.ForProvider.PreferredMaintenanceWindow != nil && !awsclient.IsEqualString(cr.Spec.ForProvider.PreferredMaintenanceWindow, cluster.PreferredMaintenanceWindow)) {
+		return false, nil
+	}
+
+	return true, nil
+}
+
+func (e *hooks) preUpdate(ctx context.Context, cr *svcapitypes.DBCluster, obj *svcsdk.ModifyDBClusterInput) error {
+	obj.DBClusterIdentifier = awsclient.String(meta.GetExternalName(cr))
+
+	if !awsclient.IsEqualString(cr.Spec.ForProvider.DBClusterIdentifier, obj.DBClusterIdentifier) {
+		obj.NewDBClusterIdentifier = cr.Spec.ForProvider.DBClusterIdentifier
+	}
```

Comment:
```suggestion
```

We cannot reliably support identifier change with how Crossplane tracks the resources today. So, I'd suggest skipping that feature of `DBCluster` API.

Reply:
Fixed. I removed changing the identifier from all other controllers as well.